### PR TITLE
[tasks] Go modules RFC release tooling

### PR DESCRIPTION
**Pull request by mx-psi**
### What does this PR do?

- Move `GoModule` to its own file
- Create `update_module` and `tag_version` tasks to respectively update inter-module dependencies and tag versions.

#### Open questions

- Should the task be used for an Agent 6 only release?
- Should `push` be on by default?

### Motivation

- Part of Go Modules RFC (RFC 779) 

### Additional Notes

- Right now `update_modules` is a no-op and `tag_version` only tags the main module with the 6 and 7 version tags.

### Describe your test plan

- For QA, during RCs use them to tag versions

### CODE REVIEW

File: tasks/release.py [hunk 1]
```diff
@@ -831,3 +838,61 @@ def create_rc(
...
+
+
+@task
+def update_modules(ctx, agent_version, verify=True):
+    """
+    Update internal dependencies between the different Agent modules.
+    * --verify checks for correctness on the Agent Version (on by default).
+
+    Examples:
+    inv -e update-modules 7.27.0
```
Comment (mx-psi) [inline tasks/release.py:860]:
```suggestion
    inv -e release.update-modules 7.27.0 
```

File: tasks/release.py [hunk 2]
```diff
@@ -14,6 +14,8 @@
 from invoke import Failure, task
 from invoke.exceptions import Exit
 
+from tasks.modules import DEFAULT_MODULES
```
Comment (mx-psi) [inline tasks/release.py:17]:
```suggestion
from .modules import DEFAULT_MODULES
```

File: tasks/release.py [hunk 3]
```diff
@@ -831,3 +840,61 @@ def create_rc(
...
+
+@task
+def tag_version(ctx, agent_version, commit="HEAD", verify=True, push=True):
+    """
+    Create tags for a given Datadog Agent version.
+    The version should be given as an Agent 7 version.
+
+    * --commit COMMIT will tag COMMIT with the tags (default HEAD)
+    * --verify checks for correctness on the Agent version (on by default).
+    * --push will push the tags to the origin remote (on by default).
```
Comment (mx-psi) [inline tasks/release.py:885]: Should this be on by default? I think it's fine (the worst that can happen is an artifact on the staging repo) but we can have it off by default.

File: tasks/modules.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        return "v0" + agent_version[1:]
+
+    def tag(self, agent_version):
+        """Return the module tag name for a given Agent version.
+        >>> mods = [GoModule("."), GoModule("pkg/util/log")]
+        >>> [mod.tag("7.27.0") for mod in mods]
+        [["6.27.0", "7.27.0"], ["pkg/util/log/v0.27.0"]]
+        """
+        if self.path == ".":
+            return ["6" + agent_version[1:], "7" + agent_version[1:]]
```
Comment (mx-psi) [inline tasks/modules.py:31]: We always return Agent 6 and Agent 7 tags. Do we want to have an option for releases that are just for one of them?

File: tasks/modules.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    def __version(self, agent_version):
+        """Return the module version for a given Agent version.
+        >>> mods = [GoModule("."), GoModule("./pkg/util/log")]
+        >>> [mod.__version("7.27.0") for mod in mods]
+        ["v7.27.0", "v0.27.0"]
+        """
+        if self.path == ".":
+            return "v" + agent_version
+
+        return "v0" + agent_version[1:]
```
Comment (albertvaka) [inline tasks/modules.py:22]: Breaks for Agent 10 :P

File: tasks/modules.py (hunk 4 above)
Comment (albertvaka) [reply tasks/modules.py:31]: We can leave it like this for now. We can change it when we decouple the release cycles of A6 and A7 in the future.

**Review by albertvaka — APPROVED** [review]
LGTM

File: tasks/modules.py (hunk 5 above)
Comment (mx-psi) [reply tasks/modules.py:22]: I think we will catch that when the time comes 😄

**Pull request merged**